### ISSUE TITLE
Pre-commit + conda on win10 looks for python in a wrong location

### ISSUE THREAD

**Issue by Alexander-Serov** [pre-commit/pre-commit#1586]
I have a problem similar to #1583, but I cannot find an easy way to fix it.

I am using conda, so my python is located in `c:\\tools\\miniconda3\\python.exe`. When I try to install the pre-commit hooks, the pre-commit file is created alright, with python pointing to the right location:
```
#!/usr/bin/env python
"""File generated by pre-commit: https://pre-commit.com"""
from __future__ import print_function

import distutils.spawn
import os
import subprocess
import sys

# work around https://github.com/Homebrew/homebrew-core/issues/30445
os.environ.pop('__PYVENV_LAUNCHER__', None)

HERE = os.path.dirname(os.path.abspath(__file__))
Z40 = '0' * 40
ID_HASH = '138fd403232d2ddd5efb44317e38bf03'
# start templated
CONFIG = '.pre-commit-config.yaml'
HOOK_TYPE = 'pre-commit'
INSTALL_PYTHON = 'c:\\tools\\miniconda3\\python.exe'
SKIP_ON_MISSING_CONFIG = False
# end templated

class EarlyExit(RuntimeError):
    pass


class FatalError(RuntimeError):
    pass
... 
```

However, when I try to commit, for some reason pre-commit is searching for python elsewhere:
```
✗ git commit
[INFO] Installing environment for https://github.com/ambv/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: Command: ('c:\\tools\\miniconda3\\python.exe', '-mvirtualenv', 
'C:\\Users\\user\\.cache\\pre-commit\\repo6nh7djva\\py_env-python3', '-p', 'C:\\python3\\python.exe')
Return code: 1
Expected return code: 0
Output:
    FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\python3\\python.exe'

Errors: (none)

Check the log at C:\Users\user/.cache\pre-commit\pre-commit.log
```

I've tried to fix it manually, but I am not sure where it takes this python location from. All references in `pre-commit` point to the correct location (including just typing `python` in the terminal). The folder `C:\\python3\\python.exe` does not exist on my system (since python is installed by conda), so it cannot be on the system path.

I will appreciate any advice, thank you.

**Comment by asottile** [comment]
why haven't you shared your configuration file?

**Comment by asottile** [comment]
(this is not at all related for #1583)

**Comment by asottile** [comment]
from searching the issue tracker for "The system cannot find the path specified" I've found these two exact duplicates:
- #1396
- #1477 

tl;dr is `language_version: python3` is trying to find python3 via the normal python ways and falling back to what you're seeing since it cannot find a suitable python3.  You can override by setting `language_version: default` in your configuration

**Comment by Alexander-Serov** [comment]
@asottile Yes, indeed I haven't found #1477. Thank you for pointing it out.

But let me just clarify the solution proposed in that issue because I am not completely sure:
- my `pre-commit` clearly indicates that the python installation is found correctly: `INSTALL_PYTHON = 'c:\\tools\\miniconda3\\python.exe'`. So it is not a question of finding python.
- I cannot install `python` in the `C:\\python3\\python.exe` location because it would conflict with the conda installation.

Could you specify what you suggest?

I would also like to specify my question:
I just want to make sure that `pre-commit` uses the python path it has detected. I don't know which line tells `pre-commit` to look in `C:\\python3\\python.exe` instead of the value of `INSTALL_PYTHON`. Could you point me to that line?

**Comment by Alexander-Serov** [comment]
I also share the configuration file you have asked for:
```
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v2.2.3
    hooks:
    -   id: check-merge-conflict
    -   id: trailing-whitespace
    -   id: end-of-file-fixer
-   repo: https://github.com/pre-commit/mirrors-isort
    rev: v4.3.21
    hooks:
    -   id: isort
-   repo: https://github.com/ambv/black
    rev: 19.10b0
    hooks:
    - id: black
```

**Comment by asottile** [comment]
one of your hooks is setting `language_version: python3` and is performing the lookup to find that (and failing)

**Comment by asottile** [comment]
you're also using a really old version of pre-commit which won't try sys.executable so maybe try upgrading

**Comment by Alexander-Serov** [comment]
Thank you!

Indeed, the problem was due to an old version of pre-commit. I didn't realize the version distributed on `conda` was not up-to-date. Once I have installed it directly from pip and updated the config, python was found correctly!
